staging/src/k8s.io/component-base/metrics/registry_test.go

```diff
@@ -515,3 +515,39 @@ func TestRegistryReset(t *testing.T) {
 		t.Fatal(err)
 	}
 }
+
+func TestDisabledMetrics(t *testing.T) {
+	SetDisabledMetric("should_be_disabled")
+	currentVersion := apimachineryversion.Info{
+		Major:      "1",
+		Minor:      "17",
+		GitVersion: "v1.17.1-alpha-1.12345",
+	}
+	registry := newKubeRegistry(currentVersion)
+	disabledMetric := NewCounterVec(&CounterOpts{
+		Name: "should_be_disabled",
+		Help: "this metric should be disabled",
+	}, []string{"label"})
+	// gauges cannot be reset
+	enabledMetric := NewGauge(&GaugeOpts{
+		Name: "should_be_enabled",
+		Help: "this metric should not be disabled",
+	})
+
+	registry.MustRegister(disabledMetric)
+	registry.MustRegister(enabledMetric)
+	disabledMetric.WithLabelValues("one").Inc()
+	disabledMetric.WithLabelValues("two").Inc()
+	disabledMetric.WithLabelValues("two").Inc()
```

Comment:
Quick question: this to me looks like very similar behaviour to how hidden metrics behave. How are we showing they are different (in the tests)?

Reply:
They don't have the deprecated help annotation in the output. You can see it in the expected output block below.